pages/admin/dashboard/builder.tsx

```diff
@@ -95,16 +97,18 @@ const Builder: Page = () => {
     const router = useRouter();
     const { moduleId } = router.query;
 
-    useEffect(() => {
-        function getUrl() {
-            return `/api/module/${moduleId}`;
-        }
-        async function fetchData() {
-            const response = await axios.get(getUrl());
-            dispatch({ type: "init", payload: response.data });
-        }
-        fetchData();
-    }, [moduleId]);
+    const fetcher = async (url) => {
+        const response = await axios({
+            method: "GET",
+            url,
+        });
+        dispatch({ type: "init", payload: response.data });
+        return response.data;
+    };
+
+    const { data, error } = useSWR(`/api/module/${moduleId}`, fetcher);
+    if (error) return "An error has occurred.";
```

Comment:
do we want a different way to indicate that an error occurred or is this message sufficient